common/utils/src/main/scala/org/apache/spark/internal/config/ConfigBuilder.scala

```diff
@@ -57,6 +57,15 @@ private object ConfigHelpers {
     }
   }
 
+  def toEnum[E <: Enum[E]](s: String, enumClass: Class[E], key: String): E = {
+    enumClass.getEnumConstants.find(_.name().equalsIgnoreCase(s.trim)) match {
+      case Some(enum) => enum
+      case None =>
+        throw new IllegalArgumentException(
+          s"$key should be one of ${enumClass.getEnumConstants.mkString(", ")}, but was $s")
```

Comment:
should we have an error condition for it?

Reply:
ditto for the scala enum

Reply:
We can have a separate PR for this to make [all](https://github.com/apache/spark/pull/50691/files#diff-60aa58f28fb7587d96dcb4031dbb86750094d9ad4a1488eb0d115b64fd3e84f0R32-R66) the condition-less errors uniform